src/NewLinkedList.java

```diff
@@ -0,0 +1,155 @@
+public class NewLinkedList<T> implements List<T> {
+	private NewNode<T> firstNewNode;
+	private NewNode<T> lastNewNode;
+	private int length = 0;
+
+	@Override
+	public void add(T value) {
+		NewNode<T> newNode = new NewNode<>(value);
+		if (isEmpty()) {
+			firstNewNode = newNode;
+			lastNewNode = newNode;
+		} else {
+			lastNewNode.setNext(newNode);
+			newNode.setPrevious(lastNewNode);
+			lastNewNode = newNode;
+		}
+		length++;
+	}
+
+	@Override
+	public void add(T value, int index) {
```

Comment:
if we write:
List<String> l = new NewLinkedList<>();
       l.add("1");
       l.add("2");
       l.add("3");
       l.add("2+", 2);
we will get:
1, 2, 2+, 3
test how work this method